### ISSUE TITLE
Feeds:  related to closed #7802

### ISSUE THREAD

**Issue by Bgolden67** [wet-boew/wet-boew#8123]
Can anyone confirm this is happening again? Our RSS feed widget on http://www.aadnc-aandc.gc.ca/prev-prev/eng/1393261289317 is not loading and doesn't seem to be loading on the example page either

**Comment by RobJohnston** [comment]
@Bgolden67 :
The [example feeds page](http://wet-boew.github.io/wet-boew/demos/feeds/feeds-en.html) seems to have a problem.  Flickr is working every time, the road news occasionally, and the weather never.

I've found that the [jQuery file from Google](https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js) isn't loading, or is slow to load, quite often lately.  Don't know if that is the issue here.

Your aadnc page gives me a "HTTP Error 403.6 - Forbidden: IP address of the client has been rejected".

**Comment by Bgolden67** [comment]
oh Crap.. i pasted the prev page.. you won't get to that... It's corrected

**Comment by RobJohnston** [comment]
Out of 10 tries, it loaded for me the 3rd time only.

**Comment by Bgolden67** [comment]
Exactly.. and it was loading fine until a few hours ago.
Which is why i think it's something to do with the yahooservers

**Comment by LaurentGoderre** [comment]
That's probably because of IP load restrictions and probably only occur in the GC or big corporate networks that report one single outside IP for all it's network.

**Comment by Bgolden67** [comment]
not so sure it's just load based. When that usually happens the links to the feed file display, now nothing displays, it's as if the widget is not there and has been doing this for a couple days now.

**Comment by MaximMartel** [comment]
@Bgolden67 Do you use OAuth with API Key? or your public ?

**Comment by Bgolden67** [comment]
@MaximMartel I don't know to be honest, here is the code..


```
<section class="wb-feeds limit-90 mrgn-tp-lg">
  <h2 class="wb-inv">Recent</h2>
  <ul class="feeds-cont lst-spcd">
  <li><a href="https://www.canada.ca/en/indigenous-northern-affairs.atom.xml" rel="external"><abbr title="Indigenous and Northern Affairs Canada">INAC</abbr> news on Canada.ca</a></li>
  </ul>
  </section>
```

**Comment by MaximMartel** [comment]
Thanks @Bgolden67 looking into this issue, update to come

**Comment by duboisp** [comment]
The issue seems to be related to the yahoo services. I am able to reproduce the same issue with the [Template HTML5 working example](http://wet-boew.github.io/themes-dist/GCWeb/demos/data-json/template-en.html) which do a custom layout from the RSS read through the yahoo API

**Comment by Bgolden67** [comment]
Is there ANY way t resolve it? or another solution? Using this feed widget has been a way to avoid manually adding the links to a New Release index page that Comms insists on having but it's causing so many issues lately - from long delays in having the display update, to not displaying at all to missing entries.
In the screen shots -  1 from Canada news search (which has a glitch of it's own) and 1 from INAC page using the widget (when it actually loads - today it is)
![can-news_search](https://user-images.githubusercontent.com/1798808/30162301-4e012788-93a2-11e7-91e0-4ff913269178.jpg)
![inac-widget](https://user-images.githubusercontent.com/1798808/30162300-4dfbd5c6-93a2-11e7-8f05-69f5c2e8ecdb.jpg)

**Comment by duboisp** [comment]
@Bgolden67 the issue look like to be more related to the availibility of the yahoo services that we use to read ATOM feed and serialize it in a JSON format. Then the ATOM feed content can be display through the feeds feature (feeds widget) or with the data-json feature.

Some solution could be:
* Take advantage of [CORS](https://www.w3.org/TR/cors/) and change the feeds feature to directly support ATOM feeds in XML.
* Take advantage of [CORS](https://www.w3.org/TR/cors/) and create an adapter "XML to JSON" in the JSON Manager feature then it will be possible to display the ATOM feed with the data-json.
* Create a web service that will replicate the yahoo services. Then that services can be used directly with the data-json to read the feeds. But the feeds features will require to get a minor update [on line 146](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/feeds/feeds.js#L146) in order to get it working.

**Comment by MaximMartel** [comment]
@Bgolden67 Is your issue still present ? Yahoo API seems to be stable now.

**Comment by Bgolden67** [comment]
Hard to tell.. at the moment it seems the widget is "working" but the feed isn't loading, however this could be a result of our firewall because that happens frequently.
When i raised this issue, nothing was loading at all, it was as if the widget wasn't even there

**Comment by thekodester** [comment]
Also on that topic it looks like Yahoo is planing on phasing it out. There website had this message:
> YQL Console
> The YQL Console is no longer available. You may continue to use YQL via our APIs. Please see the documentation here: https://developer.yahoo.com/yql/guide.

**Comment by Bgolden67** [comment]
@MaximMartel It was brought to my attention again this week that delays are frequent yet sporadic at the same time if you get my meaning.

**Comment by duboisp** [comment]
@Bgolden67 Once testing will be completed for PR #8502 it will be possible to use a feed whitelisted by the CORS http header. That should reduce a lot the delay that you are experiencing and it will offer an alternative to not depend on the Yahoo API to read a feed.

Let me know if you can help me with testing PR #8502.

**Comment by duboisp** [comment]
@Bgolden67 Thanks

I published a build that include the PR 8502. - https://github.com/wet-boew/wet-boew/pull/8502#issuecomment-440724539

The first working example use the CORS, here the link:
* Test URL: http://universallabs.org/wet/labs/pr-8502/demos/feeds/feeds-en.html
* WET-BOEW build with this fix: http://universallabs.org/wet/labs/pr-8502.zip

Let me know your testing result.
